tline-bot-algorithm/src/main/kotlin/com/betfair/tline/algorithm/HeikinAshiCandleGenerator.kt

```diff
@@ -2,15 +2,33 @@ package com.betfair.tline.algorithm
 
 import com.betfair.tline.algorithm.models.HeikinAshiCandle
 import com.betfair.tline.models.MarketPeriod
-import java.math.BigDecimal
-
-// pre-cache for performance reasons
-private val FOUR = BigDecimal(4L)
-private val TWO = BigDecimal(2L)
 
 // https://docs.trendmaster.io/essential-knowledge/heikin-ashi#construction
 class HeikinAshiCandleGenerator {
```

Comment:
Why is it called *Generator and not *Factory. Generator is more like generating someting from nothing.